Collect user timings for the timeCode function

gregtatum opened the pull request:
This helped me find the bug #1177, I think we should expose this information in our profiles as well. See the linked profile in #1177 for an example usage. It should only collect user timings in development mode.

brisad commented:
Ah, I just found out how I can see pending reviews in Github :) I had missed this one in the e-mail stream.

gregtatum commented:
Ack, apparently I moved some if branches around and messed up the timing after I recorded my profile example. I was recording the first mark after the work happened. I fixed it.
